### ISSUE TITLE
'the_content' filter called inside 'save_post' hook

### ISSUE THREAD

**Issue by hughiemolloy** [alleyinteractive/apple-news#229]
I have noticed that the '_the_content_' filter is called inside a _save_post_ action. Normally when '_the_content_' is called it is expected the global _$post_ variable is set. In the Apple News cause it is not. This causes some other plugins that edit the content to crash when they call something on a null object, like _$post->ID_.

**Comment by bcampeau** [comment]
We cannot guarantee compatibility with every plugin and if other plugins are crashing because the global `$post` object isn't set, it's because they're not implementing proper error checking. As a developer, it's not good practice to assume that data will always be valid. The usage of `the_content` there is justified to transform the post content before converting it to Apple News format. 

Which specific plugins are you having issues with? It's possible this could be an underlying cause of a couple other bug reports so we will look into this further and decide if there is an appropriate resolution.

Thanks.

**Comment by hughiemolloy** [comment]
Yes the other plugins should have error checking, but as you can see in the wordpress cpdex (https://codex.wordpress.org/Plugin_API/Filter_Reference/the_content) example the content should only be called when the global `$post` is set.

FeedWordpress and Wordpress Creation Kit to name two. This tends to happen when a third party calls `wp_insert_post()` or `wp_update_post()`.

**Comment by bcampeau** [comment]
We'll consider this for a fix in the next plugin version.
